contract_api/mpe.py

```diff
@@ -16,51 +18,81 @@ def get_latest_block_no(self):
         w3Obj = Web3(web3.providers.WebsocketProvider(self.ws_provider))
         return w3Obj.eth.blockNumber
 
-    def get_channels_by_user_address(self, user_address, org_id=None, service_id=None):
+    def get_channels_by_user_address(self, user_address, org_id=None, service_id=None, group_id=None):
         try:
-            print('Inside get_channel_info::user_address',
-                  user_address, '|', org_id, '|', service_id)
             last_block_no = self.get_latest_block_no()
-            params = [last_block_no]
-            sub_qry = ""
-            if user_address is not None and service_id is not None and org_id is not None:
-                sub_qry = " AND S.org_id = %s AND S.service_id = %s "
-                params.append(org_id)
-                params.append(service_id)
-            params.append(user_address)
-            raw_channel_dta = self.repo.execute(
-                'SELECT  C.*, E.*, IF(C.expiration > %s, "active","inactive") AS status FROM '
-                'service_group G, mpe_channel C, service_endpoint E, service S WHERE G.group_id = C.groupId AND '
-                'G.group_id = E.group_id and S.row_id = E.service_row_id and E.service_row_id = G.service_row_id '
-                'AND S.is_curated = 1  ' + sub_qry + ' AND C.sender = %s ORDER BY C.expiration DESC', params)
-            self.obj_util.clean(raw_channel_dta)
-            channel_dta = {}
-            for rec in raw_channel_dta:
-                group_id = rec['groupId']
-                if group_id not in channel_dta.keys():
-                    channel_dta[group_id] = {'group_id': group_id,
-                                             'org_id': rec['org_id'],
-                                             'service_id': rec['service_id'],
-                                             'group': {"endpoints": []},
-                                             'channels': []
-                                             }
+            params = [last_block_no, org_id, group_id, user_address]
+
+            if user_address and org_id and group_id:
+                raw_channel_data = self.repo.execute(
+                    "SELECT C.* , OG.payment, OG.org_id, IF(C.expiration > %s, 'active','inactive') AS status FROM "
+                    "mpe_channel C JOIN org_group OG ON C.groupId = OG.group_id "
+                    "where OG.org_id = %s and C.groupId = %s and C.sender = %s", params
+                )
+                self.obj_util.clean(raw_channel_data)
+                channel_data = {
+                    'group_id': group_id,
+                    'org_id': org_id,
+                    'channels': []
+                }
+                for record in raw_channel_data:
+                    record["payment"] = json.loads(record["payment"])
+                    if record["recipient"] == record["payment"]["payment_address"]:
+                        if group_id not in channel_data.keys():
```

Comment:
it could have been broken into two function which are being called from here, will enhance the readability of code